packages/block-editor/src/components/link-control/search-item.js

```diff
@@ -52,9 +52,15 @@ export const LinkControlSearchItem = ( {
 	isURL = false,
 	shouldShowType = false,
 } ) => {
+	// Remove the site URL from any suggestion that is not a manually entered URL.
+	// This is to reduce visual noise in the search results.
+	const suggestionURL = isURL
+		? suggestion?.url
+		: '/' + getPath( suggestion?.url ) + '/';
+
```

Comment:
A couple of notes:

* The `isURL` check probably isn't needed here, as the value isn't used when the condition is true.
* What do you think about extracting this into a helper function, like `getURLForDisplay`?
* We probably want to decode the URL before running it through `getPath` and `filterURLForDisplay`.
* Do we still need to use `filterURLForDisplay` here?

Reply:
Great notes. Thanks.

> The isURL check probably isn't needed here, as the value isn't used when the condition is true.

I've removed that now.

> What do you think about extracting this into a helper function, like getURLForDisplay?

I've extracted a util function which is specific to this scenario. I think it makes things more readable.

> We probably want to decode the URL before running it through getPath and filterURLForDisplay.

✅ Done. Remember `compose` applies right to left.

> Do we still need to use filterURLForDisplay here?

Yes, for the scenario where a media URL is encountered:

<img width="564" alt="Screen Shot 2023-09-19 at 09 40 30" src="https://github.com/WordPress/gutenberg/assets/444434/9f545b79-9a55-4d20-a6ec-708e8496e854">